synapseclient/client.py

```diff
@@ -5247,3 +5270,146 @@ def _return_rest_body(self, response):
         if is_json(response.headers.get("content-type", None)):
             return response.json()
         return response.text
+
+
+@dataclass
+class Permissions:
+    """
+    The permission a User has for a given Entity.
+
+    Attributes:
+        can_view (Boolean): Can the user view this entity?
+        can_edit (Boolean): Can the user edit this entity?
+        can_move (Boolean): (Read Only) Can the user move this entity by changing its parentId?
+        can_add_child (Boolean): Can the user add a child entity to this entity?
+        can_certified_user_edit (Boolean): (Read Only) Can the user edit this entity once they become a Certified User?
+        can_certified_user_add_child (Boolean): (Read Only) Can the user add a child entity to this entity once they become a Certified User?
+        is_certified_user (Boolean): (Read Only) True, if the user has passed the user certification quiz.
+        can_change_permissions (Boolean): Can the user change the permissions of this entity?
+        can_change_settings (Boolean): Can the user change the settings of this entity?
+        can_delete (Boolean): Can the user delete this entity?
+        can_download (Boolean): Are there any access requirements precluding the user from downloading this entity?
+        can_upload (Boolean): (Read Only) Are there any access requirements precluding the user from uploading into this entity (folder or project)?
+        can_enable_inheritance (Boolean): (Read Only) Can the user delete the entity's access control list (so it inherits settings from an ancestor)?
+        owner_principal_id (Integer): (Read Only) The principal ID of the entity's owner (i.e. the entity's 'createdBy')
+        can_public_read (Boolean): (Read Only) Is this entity considered public?
+        can_moderate (Boolean): Can the user moderate the forum associated with this entity? Note that only project entity has forum.
+        is_certification_required (Boolean): (Read Only) Is the certification requirement enabled for the project of the entity?
+        is_entity_open_data (Boolean): (Read Only) Returns true if the Entity's DateType equals 'OPEN_DATA', indicating that the data is safe to be released to the public.
+    """
+
+    can_view: bool = None
+    """Can the user view this entity?"""
```

Comment:
Formatting might look a little better to add a new line inbetween the docstring and the new attribute below.